### PR TITLE
fix(battery_plus): Fix type conversion

### PR DESCRIPTION
**Pull request by koji-1009**
## Description

When wasm is used, there is a complete distinction between double and int. For this reason, casting a double type to an int type with `as` will cause a crash in the wasm environment. Fix this problem by changing from `as int` to `toInt()`.

---

https://github.com/flutter/flutter/pull/146231

Starting with flutter 3.24, the `flutter run --wasm` command makes it easy to try out the wasm version of the app at local.

## Related Issues

- Related https://github.com/fluttercommunity/plus_plugins/pull/3103#issuecomment-2273891732

## Checklist

- [x] I read the [Contributor Guide](https://github.com/fluttercommunity/plus_plugins/blob/main/CONTRIBUTING.md) and followed the process outlined there for submitting PRs.
- [x] I titled the PR using [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0).
- [x] I did not modify the `CHANGELOG.md` nor the plugin version in `pubspec.yaml` files.
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate that with a `!` in the title as explained in [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0)).
- [x] No, this is *not* a breaking change.

### CODE REVIEW

**Comment by miquelbeltran** [comment]
Makes sense!

can you rebase this PR to https://github.com/fluttercommunity/plus_plugins/pull/3103 ?

I'd also prefer if the PR only contained the `toInt` change and not the other files.

Thanks!

**Comment by koji-1009** [comment]
@miquelbeltran 
I've finished rebasing and organizing the commits!

**Comment by miquelbeltran** [comment]
Thank you! I will merge this one once #3103 lands

**Review by miquelbeltran — APPROVED** [review]

**Comment by miquelbeltran** [comment]
ugh why is closed...

**Comment by miquelbeltran** [comment]
For some reason GH closed this automatically instead of changing the target to `main`, reopening here https://github.com/fluttercommunity/plus_plugins/pull/3157

**Comment by koji-1009** [comment]
No problem! Thanks Marge!

**Pull request closed without merging**
